### PR TITLE
[WIP] feat: InfoTip Component

### PR DESCRIPTION
**Pull request by aljesusg**
This is for Infotip component http://www.patternfly.org/pattern-library/widgets/#info-tip
Issue #170 
I think that is not necessary create a component we only need styles so I add this to the storybook. What do you think?
captura de pantalla 2018-01-24 a las 9 05 26

Anyway After merge of Nav this code will be more cleaner

### CODE REVIEW

**Comment by sharvit** [comment]
Thanks @aljesusg, I won't agree with you here, I think you can create some very useful components.

I would love to consume it similar to this:
```jsx
<InfoTipDropdown>
  <InfoTipDropdown.Toggle>Messages: 2</InfoTipDropdown.Toggle>
  <InfoTipDropdown.Menu>
    <ListGroup>
      <ListGroupItem>
        <Icon type="pf" name="info" /> Added Datasources TestDS
      </ListGroupItem>
      <ListGroupItem>
        <Icon type="pf" name="info" /> Modified Datasources ExampleDS
      </ListGroupItem>
    </ListGroup>
    <InfoTipDropdown.MenuFooter>
      <a href="#">Clear Messages</a>
    </InfoTipDropdown.MenuFooter>
  </InfoTipDropdown.Menu>
</InfoTipDropdown>
```

What do you think? See the benefit?

**Comment by aljesusg** [comment]
Thanks @sharvit I was waiting some comment about, I didn't know if you wanna a specific component for this or not because with the classes we can have the same result so I'll do the changes.

**Comment by sharvit** [comment]
I think one of the things make react so great is we can make the classes transparent to use,
so as a consumer, instead of integrating with classes (which is hard to follow) I can integrate with components and their props.

I embrace creating dummy components as this one:
```jsx
const SomeComponent = ({ children }) => <div className="some-class">{children}</div>;
```

**Comment by aljesusg** [comment]
Ok I understand... I didn't know  how you work ^^ Thanks @sharvit

**Comment by aljesusg** [comment]
Hi @sharvit, I have a problem when I separate Dropdown Toogle from Menu **Cannot read property 'focusNext' of undefined** is related with 
```
Dropdown.prototype.focusNextOnOpen = function focusNextOnOpen() {
    var menu = this.menu;

    if (!menu.focusNext) {
```
The point is that Dropdown don't find the menu, so I don't know if use the DOM to find the element. Sorry but I have no much experience in react. 

With this throw the error
```react-js
              <InfoTip>  
                <InfoTip.Toggle bsRole="toggle">
                   Messages: 2
                </InfoTip.Toggle>
                <InfoTip.Menu bsRole="menu">
                    <ListGroup>
                        <ListGroupItem>
                        <Icon type="pf" name="info" /> Added Datasources TestDS
                        </ListGroupItem>
                        <ListGroupItem>
                        <Icon type="pf" name="info" /> Modified Datasources ExampleDS
                        </ListGroupItem>
                    </ListGroup>
                    <InfoTip.MenuFooter>
                        <a href="#">Clear Messages</a>
                    </InfoTip.MenuFooter>
                </InfoTip.Menu>    
              </InfoTip>            
```

**Comment by sharvit** [comment]
@aljesusg, sorry for the delay but I haven't forgotten you, will try to jump and help soon.

**Comment by aljesusg** [comment]
Don't worry @sharvit, I'll start another PR while I wait.

**Comment by sharvit** [comment]
Hi @aljesusg, check this out, I converted your components to a `class` components and it works :+1: 

```jsx
class Menu extends React.Component {
  render() {
    const { children, ...props } = this.props;

    return (
      <Dropdown.Menu className="infotip bottom-right" {...props}>
        <div className="arrow" />
        {children}
      </Dropdown.Menu>
    );
  }
}
```

I think you should migrate all your components to this syntax.

When implementing class components, your component will have the react-lifecycle:
https://reactjs.org/docs/react-component.html
